multicluster/controllers/multicluster/member/node_controller.go

```diff
@@ -324,12 +373,47 @@ func (r *NodeReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	}
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Node{}).
+		Watches(&source.Kind{Type: &mcv1alpha2.ClusterSet{}},
+			handler.EnqueueRequestsFromMapFunc(r.clusterSetMapFunc),
+			builder.WithPredicates(statusReadyPredicate)).
 		WithOptions(controller.Options{
 			MaxConcurrentReconciles: 1,
 		}).
 		Complete(r)
 }
 
+func (r *NodeReconciler) clusterSetMapFunc(a client.Object) []reconcile.Request {
+	clusterSet := &mcv1alpha2.ClusterSet{}
+	requests := []reconcile.Request{}
+	if a.GetNamespace() != r.namespace {
+		return requests
+	}
+	ctx := context.TODO()
+	err := r.Client.Get(ctx, types.NamespacedName{Namespace: a.GetNamespace(), Name: a.GetName()}, clusterSet)
+	if err == nil {
+		if len(clusterSet.Status.Conditions) > 0 && clusterSet.Status.Conditions[0].Status == corev1.ConditionTrue {
+			nodeList := &corev1.NodeList{}
+			r.Client.List(ctx, nodeList)
+			for _, n := range nodeList.Items {
+				if _, ok := n.Annotations[common.GatewayAnnotation]; ok {
+					requests = append(requests, reconcile.Request{
+						NamespacedName: types.NamespacedName{
+							Name: n.GetName(),
+						},
+					})
+				}
+			}
+		}
+	} else if apierrors.IsNotFound(err) {
+		r.activeGatewayMutex.Lock()
+		defer r.activeGatewayMutex.Unlock()
+		// The Gateway will be removed when a ClusterSet is deleted, so here we can set
+		// the activeGateway to empty directly.
+		r.activeGateway = ""
```

Comment:
I don't see how the gateway will be removed when a ClusterSet is deleted, especially when the activeGateway is reset here.
I think updating activeGateway should be done in `Reconcile` and the map func should only trigger its process.

Reply:
Comment added.